src/PrestaShopBundle/Controller/Admin/Improve/International/CurrencyController.php

```diff
@@ -462,6 +464,67 @@ private function getSettingsFormHandler()
         return $this->get('prestashop.admin.currency_settings.form_handler');
     }
 
+    /**
+     * Toggles currencies status in bulk action
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", redirectRoute="admin_currencies_index")
+     * @DemoRestricted(redirectRoute="admin_currencies_index")
+     *
+     * @param Request $request
+     * @param string $status
+     *
+     * @return RedirectResponse
+     */
+    public function bulkToggleStatusAction(Request $request, $status)
+    {
+        $currenciesIds = $this->getBulkCurrenciesFromRequest($request);
+        $expectedStatus = 'enable' === $status;
+
+        try {
+            $this->getCommandBus()->handle(new BulkToggleCurrenciesStatusCommand(
+                $currenciesIds,
+                $expectedStatus
+            ));
+
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (CurrencyException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
```

Comment:
You should update `getErrorMessageForException` to manage the new exception code you added